src/tir/usmp/unified_static_memory_planner.cc

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file tir/analysis/usmp/unified_static_memory_planner.cc
+ * \brief This is the pass that integrates the USMP passes to
+ * a single composite pass.
+ */
+
+#include <tvm/target/target.h>
+#include <tvm/tir/stmt_functor.h>
+#include <tvm/tir/transform.h>
+#include <tvm/tir/usmp/algo/algo.h>
+#include <tvm/tir/usmp/analysis.h>
+#include <tvm/tir/usmp/transform.h>
+#include <tvm/tir/usmp/utils.h>
+
+#include <string>
+
+namespace tvm {
+
+TVM_REGISTER_PASS_CONFIG_OPTION("tir.usmp.enable", Bool);
+TVM_REGISTER_PASS_CONFIG_OPTION("tir.usmp.algorithm", String);
+
+namespace tir {
+namespace usmp {
+
+static constexpr const char* kDefaultAlgo = "greedy_by_size";
+
+static std::unordered_map<String, std::function<Map<BufferInfo, PoolAllocation>(
+                                      const Array<BufferInfo>&, const Integer&)>>
+    algorithms{{"greedy_by_size", algo::GreedyBySize},
+               {"greedy_by_conflicts", algo::GreedyByConflicts}};
+
+IRModule PlanMemory(const IRModule& mod, String algo) {
+  VLOG(1) << "workspace required = " << CalculateModuleWorkspaceSize(mod);
+  PrimFunc main_func = Downcast<PrimFunc>(mod->Lookup(::tvm::runtime::symbol::tvm_run_func_suffix));
+  BufferInfoAnalysis buffer_info_analysis = ExtractBufferInfo(main_func, mod);
+  Array<BufferInfo> buffer_info_arr =
+      CreateArrayBufferInfo(buffer_info_analysis->buffer_info_stmts);
+  Map<BufferInfo, PoolAllocation> buffer_info_pool_allocations =
+      algorithms[algo](buffer_info_arr, buffer_info_analysis->memory_pressure);
```

Comment:
suggest a diagnostic for invalid algorithm tag

Reply:
Added that!